services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/mapper/annotations/DynamoDbPreserveEmptyBean.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.enhanced.dynamodb.mapper.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.util.Map;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.enhanced.dynamodb.TableSchema;
+
+/**
+ * Class level annotation that specifies that when calling {@link TableSchema#mapToItem(Map)},
+ * this class should be initialized as empty class if all fields in this class are null
+ *
+ * <p>
+ * eg:
+ * <pre>
+ *     {@code
+ * @DynamoDbBean
+ * @DynamoDbPreserveEmptyBean
```

Comment:
I'm not feeling good seeing how this looks. Have we considered something like @DynamoDbBean(preserveEmptyBean = true) as an alternative?

Reply:
I thought about it; I was not sure if this was the best way to do it because we'd need to add it to all class-level bean annotations, `DynamoDbBean` and `DynamoDbImmutable`, for now and new bean annotations in the future. Using the annotation attribute makes sense since preserveEmpty bean is like a configuration on the bean.

Reply:
To close the conversation, discussed offline. Changed to use method annotation instead.